src/core/link-to-dfn.js

```diff
@@ -46,6 +44,20 @@ export function run(conf, doc, cb) {
         }
       }
     });
+    if (listOfDuplicateDfns.length > 0) {
```

Comment:
Ok, we want to refactor this to use functional style... so, something like:

```JS
const dfnsList = listOfDuplicateDfns.map((elem, i) => {
  return `[${i + 1}](#${elem.id})`;
}).join(", ");
```

Reply:
Also just realized that we want to add the CSS class "respec-offending-element" to any duplicates, so that duplicate definitions are easier to spot. We might also want to set the `title` attribute of the element to "This is defined more than once in the document".

Reply:
Same for here also. As this will fail when I have two or more titles with duplicate deinitions for which the dfnsList is reconstructed every time for each and every `title`.

Reply:
> Also just realized that we want to add the CSS class "respec-offending-element" to any duplicates, so that duplicate definitions are easier to spot. We might also want to set the title attribute of the element to "This is defined more than once in the document".

On it

Reply:
Yes in a minute

Reply:
I have made all the changes except some changes related to setting the variable as `const`. I may be wrong also, in this case I'm sincerely sorry

`I have created big spaces to show where I have made the code change`
```javascript
  Object.keys(conf.definitionMap).forEach(function(title) {
    titles[title] = {};


    var listOfDuplicateDfns = []; /*
     Now as you can see this array has to get updated every time the title changes. As this stores 
     the list of duplicate dfns. So how can I set this as const as this will be modified a number of 
     times for the same title if there are two or more duplicate dfns. **This is the only place where I haven't followed your review. Other wise everywhere below you can find the changes as per your review** */


    conf.definitionMap[title].forEach(function(dfn) {
      if (dfn.attr("data-idl") === undefined) {
        // Non-IDL definitions aren't "for" an interface.
        dfn.removeAttr("data-dfn-for");
      }
      var dfn_for = dfn.attr("data-dfn-for") || "";
      if (dfn_for in titles[title]) {
        // We want <dfn> definitions to take precedence over
        // definitions from WebIDL. WebIDL definitions wind
        // up as <span>s instead of <dfn>.
        var oldIsDfn = titles[title][dfn_for].filter("dfn").length !== 0;
        var newIsDfn = dfn.filter("dfn").length !== 0;
        if (oldIsDfn && newIsDfn) {
          // Only complain if the user provides 2 <dfn>s
          // for the same term.


          dfn.addClass("respec-offending-element");
          dfn.attr("title", "This is defined more than once in the document");
          /* I have made the changes here as you had requested*/


         if (dfn.attr("id") === undefined) {
            dfn.makeID(null, title);
          }


          listOfDuplicateDfns.push(dfn[0]);
          /*Made this change too*/
 
       }
        if (oldIsDfn) {
          // Don't overwrite <dfn> definitions.
          return;
        }
      }
      titles[title][dfn_for] = dfn;
      if (dfn.attr("id") === undefined) {
        if (dfn.attr("data-idl")) {
          dfn.makeID("dom", (dfn_for ? dfn_for + "-" : "") + title);
        } else {
          dfn.makeID("dfn", title);
        }
      }
    });
    if (listOfDuplicateDfns.length > 0) {
      
     
      const dfnsList = listOfDuplicateDfns.map((elem, i) => {
        return `[${i + 1}](#${elem.id})`;
      }).join(", ");
     /* incorporated the changes here */

      pub(
        "error",


        `Duplicate definitions of '${title}' at: ${dfnsList}.`
         /* made the changes here too as per your request.*/

      );
    }
  });
```

Reply:
Ah, thanks! got it 👍 You are right.

Reply:
Looks like this module a bigger refactor... but that's for another day :)

Reply:
So shall I update this PR? Here is a GIF:

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/30361728/35901969-6c6e7646-0bff-11e8-84a1-8abc20b1e388.gif)
___
> Looks like this module a bigger refactor... but that's for another day :)

I would love to do this!

Reply:
Awwweesome! yeah, send up the new code!

Reply:
Sending in a minute